flyteplugins/go/tasks/plugins/webapi/agent/client.go

```diff
@@ -147,15 +147,21 @@ func initializeAgentRegistry(cs *ClientSet) (Registry, error) {
 				agent := &Agent{AgentDeployment: agentDeployment, IsSync: agent.IsSync}
 				agentRegistry[supportedCategory.GetName()] = map[int32]*Agent{supportedCategory.GetVersion(): agent}
 			}
-			logger.Infof(context.Background(), "[%v] is a sync agent: [%v]", agent.Name, agent.IsSync)
-			logger.Infof(context.Background(), "[%v] supports task category: [%v]", agent.Name, supportedTaskCategories)
+		}
+		// If the agent doesn't implement the metadata service, we construct the registry based on the configuration
+		for taskType, agentDeploymentID := range cfg.AgentForTaskTypes {
+			agentDeployment, ok := cfg.AgentDeployments[agentDeploymentID]
+			if ok {
+				agent := &Agent{AgentDeployment: agentDeployment, IsSync: false}
+				agentRegistry[taskType] = map[int32]*Agent{defaultTaskTypeVersion: agent}
+			}
 		}
 	}
-
-	return agentRegistry, nil
+	logger.Debugf(context.Background(), "AgentDeployment service supports task types: %v", maps.Keys(agentRegistry))
```

Comment:
```suggestion
	logger.Debugf(ctx, "AgentDeployment service supports task types: %v", maps.Keys(agentRegistry))
```
Should we use existing context?

Reply:
done